src/components/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,156 @@
+import React, {useCallback, useState, useMemo} from 'react'
+import {Text, FlatList, StyleProp, ViewStyle, TextStyle} from 'react-native'
+import AccordionItem from './AccordionItem'
+import type {AnimationType} from './ToggleAnimation'
+
+type ViewStyleProp = StyleProp<ViewStyle> | Array<StyleProp<ViewStyle>>
+type TextTitleStyleProp = StyleProp<TextStyle> | Array<StyleProp<TextStyle>>
+
+export type section = {
+  [key: string]: any
+  title: string
+  content: string
+}
+
+export interface CommonAccordionProps {
+  /**
+   * onPress accordion event
+   */
+  onPress?: (keyExtractorItem: string) => void
+  /**
+   * render custom accordion header
+   */
+  renderHeader?: (
+    item: section,
+    index: number,
+    expanded: boolean,
+    keyExtractorItem: string,
+  ) => React.ReactNode
+  /**
+   * render custom title section
+   */
+  renderSectionTitle?: (item: section, index: number, expanded: boolean) => React.ReactNode
+  /**
+   * render custom content
+   */
+  renderContent?: (
+    item: section,
+    index: number,
+    expanded: boolean,
+    keyExtractorItem: string,
+  ) => React.ReactNode
+  /**
+   * accordion title
+   */
+  title?: string
+  /**
+   * accordion container style
+   */
+  itemContainerStyle?: ViewStyleProp
+  /**
+   * accordion title style
+   */
+  titleStyle?: TextTitleStyleProp
+  /**
+   * accordion header style
+   */
+  headerContainerStyle?: ViewStyleProp
+  /**
+   * accordion body style
+   */
+  contentContainerStyle?: ViewStyleProp
+  /**
+   * animation type when open
+   * default: easeInEaseOut
+   */
+  openAnimation?: AnimationType
+  /**
+   * animation type when close
+   * default: easeInEaseOut
+   */
+  closeAnimation?: AnimationType
+  /**
+   * duration when open
+   * default: 300
+   */
+  openDuration?: number
+  /**
+   * duration when close
+   * default: 300
+   */
+  closeDuration?: number
+}
+
+export interface AccordionProps extends CommonAccordionProps {
+  /**
+   * section data sets
+   */
+  sections: section[]
+  /**
+   * keyExtractor for Flatlist
+   */
+  keyExtractor?: (item: section, index: number) => string
+  /**
+   * enable expand multiple item
+   * default: false
+   */
+  expandMultiple?: false
+  /**
+   * accordion wrapper style
+   */
+  wrapperStyle?: ViewStyleProp
+}
+
+const Accordion = React.forwardRef<FlatList, AccordionProps>(
+  ({sections, expandMultiple = true, keyExtractor, wrapperStyle, ...props}, ref) => {
```

Comment:
I'm confuse with **expandMultiple**
In line 95, I see comment **default: false** for props expandMultiple but see you set default **expandMultiple = true** in this line

Reply:
updated